### PR TITLE
SSH Agent: Add support for generating SSH keys

### LINKED ISSUE #1761: SSH key generation
Add "Generate SSH key" button. This will resolve (partly) issues with key formats.

## Expected Behavior

Add button "Generate key" between Private key label and Attachment option, which allows user to generate new key. It will generate attachment.
image

## Possible Solution

As an alternative, you can add third option to "Private key" section, which launch little wizard with some fields like "Comment", "Format type", etc...

## Context

In real world ill be using these keys to connect to servers (one for each server), so having option to generate right inside program is a plus for me.

### PR DESCRIPTION
**Pull request by hifi**
Supported key types are RSA, ECDSA and Ed25519.

Includes tests to compare writing out keys produce the exact same
private key if read from OpenSSH format and tests against ssh-agent
to ensure all no generated key is rejected.

Fixes #1761 

[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )

## Screenshots
The "Generate" button has been added right next to the attachment list which pop up the dialog. Generated key will be automatically selected when you click OK.

image

## Testing strategy
Tested key generation on Linux and the keys seem to work, needs more testing, though.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (change that adds functionality)

### CODE REVIEW

**Comment by lammel** [comment]
If we are able to generate a private key, an option to export the private key would be very useful. Otherwise it is quite hard to work with the key externally (or move the key out of keepassxc again).
A button "Export private key" could be added just next to "Generate" or "Remove from agent".

**Comment by hifi** [comment]
The generated key is saved as an attachment and it's readily exportable like any other attached file.

File: src/sshagent/OpenSSHKey.cpp [hunk 1]
```diff
@@ -135,6 +196,11 @@ void OpenSSHKey::setType(const QString& type)
     m_type = type;
 }
 
+void OpenSSHKey::setCheck(quint32 check)
+{
+    m_check = check;
+}
```
Comment (droidmonkey) [inline src/sshagent/OpenSSHKey.cpp:203]: This function name and purpose is obtuse. What is meant by check? What happens if I don't call this function? We don't do any checks in the privateKey() function to ensure the value is valid.

Comment (hifi) [reply src/sshagent/OpenSSHKey.cpp:203]: It's just a random integer that is tested to check the integrity of a decrypted key. I just used the same variable name as OpenSSH developers :sweat_smile: 

https://github.com/openssh/openssh-portable/blob/master/sshkey.c#L3958

**Comment by droidmonkey** [comment]
@hifi do you want to kick this out of draft?

**Comment by droidmonkey** [comment]
@hifi is this ready to go?

**Comment by hifi** [comment]
Uh, right. If nothing sticks out rebase + merge would work. One known issue is key generation blocks UI which is mostly only a problem with RSA keys. If that's not a blocker then this could be merged as an MVP.

**Review by droidmonkey — APPROVED** [review]

**Pull request merged**